src/dapp/src/shared/csvSigner.ts

```diff
@@ -7,24 +7,35 @@ interface SignedCsv {
   signature: string
 }
 
-export async function signCsv(csv: string): Promise<{ signedCsv: SignedCsv[]; publicKey: Uint8Array }> {
+export type SnapshotRow = {
+  address: string
+  weight?: number
+}
+
+export async function signCsv(csv: string, includeWeighting: boolean): Promise<{ signedCsv: SignedCsv[]; publicKey: Uint8Array }> {
   if (!window.isSecureContext) {
     alert('Page not in a secure context so aborting to avoid potential leak of private signing key')
     throw new Error('Page not in a secure context so aborting to avoid potential leak of private signing key')
   }
 
-  const results = Papa.parse(csv)
+  const results = Papa.parse<SnapshotRow>(csv, { header: true })
   const privateKey = ed.utils.randomPrivateKey()
   const publicKey = await ed.getPublicKeyAsync(privateKey)
   const signedCsv = await Promise.all(
     results.data
-      .filter((row) => !Array.isArray(row) || !!row[0])
+      .filter((row) => !Array.isArray(row) || !!row.address)
```

Comment:
Should only check the address here. Given we are using headers the `row` is an object where the headers are the keys